gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GitFlowGraphMonitor.java

```diff
@@ -127,6 +128,14 @@ public boolean shouldPollGit() {
    */
   @Override
   void processGitConfigChanges() throws GitAPIException, IOException {
+    if (flowTemplateCatalog.isPresent() && (flowTemplateCatalog.get() instanceof ObservingFSFlowEdgeTemplateCatalog)) {
```

Comment:
We shouldn't be doing this check on every processGitCOnfigChanges() call.  Can't we just have a method isShouldRefreshFlowGraph() in the parent Catalog class that returns false and have this method overriden in the observing catalog child class?

Reply:
I didn't do this since refresh flow graph only makes sense for observing flow catalog, but yeah I guess doing instanceof is not great either so I'll change it.